### ISSUE TITLE
False positive missing unsafe on WASM32 intrinsics

### ISSUE THREAD

**Issue by eternal-flame-AD** [rust-lang/rust-analyzer#20640]
Looks similar to #5412  but doesn't look like it is the same issue (no proc_macros involved).

**rust-analyzer version**: rust-analyzer version: 0.4.2608-standalone

**rustc version**: (eg. output of `rustc -V`): Reproducible on:
  - rustc 1.91.0-nightly (9c27f27ea 2025-09-08)
  - rustc 1.89.0 (29483883e 2025-08-04)

**editor or extension**: VSCode (reproducible on Cursor as well) 0.4.2608

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

```json
{
    "rust-analyzer.cargo.target": "wasm32-unknown-unknown"
}
```

**code snippet to reproduce**:
```rust
use core::arch::wasm32::*;

fn get_hello() -> v128 {
    u32x4(
        u32::from_ne_bytes(*b"hell"),
        u32::from_ne_bytes(*b"o,wo"),
        u32::from_ne_bytes(*b"rld!"),
        0,
    )
}

fn main() {
    let mut out = [0u8; 16];
    unsafe {
        let hello = get_hello();
        v128_store(out.as_mut_ptr().cast(), hello);
    };
    println!("{:?}", String::from_utf8_lossy(&out));
}


```

Output:

```
[{
	"resource": "rs-play/src/main.rs",
	"owner": "rust-analyzer",
	"code": {
		"value": "E0133",
		"target": {
			"$mid": 1,
			"path": "/stable/error_codes/E0133.html",
			"scheme": "https",
			"authority": "doc.rust-lang.org"
		}
	},
	"severity": 8,
	"message": "call to unsafe function is unsafe and requires an unsafe function or block",
	"source": "rust-analyzer",
	"startLineNumber": 4,
	"startColumn": 5,
	"endLineNumber": 9,
	"endColumn": 6,
	"modelVersionId": 5
}]
```

But it builds and runs, on Rustdoc this function is not marked unsafe:

```sh
> cargo run --release --target wasm32-wasip1 
     Running `wasmtime target/wasm32-wasip1/release/rs-play.wasm`
"hello,world!\0\0\0\0"
```

I tried combinations of -Ctarget-feature=+simd128 or no target-features, wasm32-unknown-unknown and wasm32-wasip1, the error still persists.

**Comment by ChayimFriedman2** [comment]
Ooh, on WASM specifically safe `#[target_feature]` functions can always be called 🤔

**Comment by eternal-flame-AD** [comment]
yeah, it's different from other platforms for some reason where sets are still marked unsafe.

Currently I wrap the code in unsafe with allow(unused_unsafe) to workaround

**Comment by ChayimFriedman2** [comment]
FWIW you can add `E0133` to [rust-analyzer.diagnostics.disabled](https://rust-analyzer.github.io/book/configuration.html#diagnostics.disabled) to disable the unsafe diagnostic (always).
